backend/question-service/src/questions/questions.controller.ts

```diff
@@ -1,34 +1,95 @@
-import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  Query,
+  NotFoundException,
+  BadRequestException,
+  InternalServerErrorException,
+} from '@nestjs/common';
 import { QuestionsService } from './questions.service';
 import { CreateQuestionDto } from './dto/create-question.dto';
 import { UpdateQuestionDto } from './dto/update-question.dto';
+import { Question } from './schemas/question.schema';
+import { QuestionCategory, QuestionComplexity } from './types/question.types';
+import { FilterQuestionDto } from './dto/filter-question.dto';
 
 @Controller('questions')
 export class QuestionsController {
   constructor(private readonly questionsService: QuestionsService) {}
 
   @Post()
-  create(@Body() createQuestionDto: CreateQuestionDto) {
-    return this.questionsService.create(createQuestionDto);
+  async create(@Body() createQuestionDto: CreateQuestionDto): Promise<Question> {
+    try {
+      return await this.questionsService.create(createQuestionDto);
+    } catch (error) {
+      throw new InternalServerErrorException(`Failed to create question`);
+    }
   }
 
   @Get()
-  findAll() {
-    return this.questionsService.findAll();
+  async findAll(
+    @Query('categories') categories?: QuestionCategory[],
+    @Query('complexity') complexity?: QuestionComplexity
+  ) {
+    try {
+      const filterDto: FilterQuestionDto = {};
```

Comment:
Perhaps we should consider modifying the FilterQuestionDto to only have categories and complexity instead of it being a general subset of the CreateQuestionDto